flyteadmin/pkg/manager/impl/util/shared.go

```diff
@@ -15,17 +15,23 @@ import (
 	repoInterfaces "github.com/flyteorg/flyte/flyteadmin/pkg/repositories/interfaces"
 	"github.com/flyteorg/flyte/flyteadmin/pkg/repositories/models"
 	"github.com/flyteorg/flyte/flyteadmin/pkg/repositories/transformers"
+	"github.com/flyteorg/flyte/flyteadmin/pkg/runtime"
 	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/admin"
 	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/core"
 	"github.com/flyteorg/flyte/flytestdlib/logger"
 	"github.com/flyteorg/flyte/flytestdlib/storage"
 )
 
-func GetExecutionName(request admin.ExecutionCreateRequest) string {
+func GetExecutionName(request admin.ExecutionCreateRequest) (string, error) {
 	if request.Name != "" {
-		return request.Name
+		return request.Name, nil
 	}
-	return common.GetExecutionName(time.Now().UnixNano())
+	config := runtime.NewApplicationConfigurationProvider()
+	executionName, err := common.GetExecutionName(time.Now().UnixNano(), config.GetTopLevelConfig().FeatureGates.EnableHumanHash)
+	if err != nil {
+		return "", err
+	}
+	return executionName, nil
```

Comment:
we shouldn't need this once we go back to the original signature for `GetExecutionName`.